### PR TITLE
Multi Tag Subscription

### LINKED ISSUE #3931: Combined Subscribe and login flow
Combined “Subscribe and login” flow - get people subscribed to some tags as they sign up, based on tags from the pages they've been looking at
    * a route or parameters or session modification to pass some tag names in upon logging in, which then are auto-subscribed to with a notification and link to /subscriptions
Refer #3924

### PR DESCRIPTION
**Pull request by SidharthBansal**
Fixes #3931 (<=== Add issue number here)

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [ ] code is in uniquely-named feature branch and has no merge conflicts
* [ ] PR is descriptively titled
* [ ] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

**Comment by SidharthBansal** [comment]
Initially
image

On http://localhost:3000/subscribe/multiple_tag/tag/kite
image

On http://localhost:3000/subscribe/multiple_tag/tag/kites,balloon
image

Everything working fine.

**Comment by SidharthBansal** [comment]
@jywarren please review this and suggest changes.

File: config/routes.rb [hunk 1]
```diff
@@ -86,6 +86,8 @@
   get 'subscribe/:type/:name' => 'subscription#add'
   get 'subscriptions' => 'subscription#index'
   get 'subscriptions/digest' => 'subscription#digest'
+  get 'subscribe/multiple_tag/:type/:names' => 'subscription#multiple_add'
```
[the remark this answers is not in the document]
Comment (SidharthBansal) [reply config/routes.rb]: I have never used the query strings. I was not sure about how to pass a different number of tags in the route at different times. So I created this route instead. 
The string can be splitted into multiple tags and we can achieve the same functionality.

**Comment by SidharthBansal** [comment]
@jywarren I think we can dismiss the codeclimate

**Comment by SidharthBansal** [comment]
@gauravano can you please review this while Jeff is busy so that we can correct anything which I have missed?
Thanks

**Comment by jywarren** [comment]
maybe the route could be "multiple" instead of "multiple_tag" which is a
little redundant?

Will review soon but this is looking great!!!

**Comment by SidharthBansal** [comment]
>>maybe the route could be "multiple" instead of "multiple_tag" which is a
little redundant?

I have pushed the requested changes.

**Comment by SidharthBansal** [comment]
Yeah sure!!!
Let's first continue #4340. Once @jywarren will review and I will complete
the suggested changes and it gets merged.
You can continue on this.
I hoped that that pr will be merged some days before. Sorry Isha to trouble
you.

**Comment by IshaGupta18** [comment]
Oh please, you haven't troubled me at all. Please take all the time you need. You all have been really cooperative with me, thank you for that, and you haven't caused me any trouble whatsoever.

**Comment by SidharthBansal** [comment]
I will add it in the follow up PR. Keeping PRS modular as per your suggestion.
Actually, I prepared tests in my rough notebook. Can't find the subscription file to write the tests. Can you please tell the file? I will send the patch quickly. Do I need to write tests in unit/node_tag?

**Comment by SidharthBansal** [comment]
As it is approved by Jeff. So, merged it.
Please tell the file in which I need to write functional tests @jywarren .Sorry I can't find it.

**Comment by jywarren** [comment]
At this point most of the new code is in the controller so let's stick with
a test in /test/functional/... Alongside the other tests for actions in the
same controller. But I love unit tests because they're more universal and
take less resources generally! So we could think about that later maybe.

**Comment by SidharthBansal** [comment]
Thanks, I will write tests in plots2/test/functional/subscription_controller_test.rb.
Some amazing work going on with help of @IshaGupta18 in #4344. Check it out whenever you get some time.
Thanks

**Comment by jywarren** [comment]
Awesome I will!!!

**Pull request merged**